matrix-room.c

```diff
@@ -1163,6 +1237,30 @@ void matrix_room_send_image(PurpleConversation *conv, int imgstore_id,
 }
 
 /**
+ * Sends a typing notification in a room with a 25s timeout
+ */
+void matrix_room_send_typing(PurpleConversation *conv, gboolean typing)
+{
+    JsonObject *content;
+    MatrixConnectionData *acct;
+    PurpleConnection *pc = conv->account->gc;
+
+    acct = purple_connection_get_protocol_data(pc);
+    
+    content = json_object_new();
+    json_object_set_boolean_member(content, "typing", typing);
+    if (typing == TRUE) {
+        json_object_set_int_member(content, "timeout", 25000);
+    }
+    
+    // Don't check callbacks as it's inconsequential whether typing notifications go through
```

Comment:
setting the callbacks to NULL isn't the same as not checking them, fwiw - they have defaults (which IIRC disconnect the account). That's probably not important, though.